src/roiextractors/extraction_tools.py

```diff
@@ -26,6 +29,88 @@
 FloatType = float
 
 
+@dataclass()
+class VideoStructure:
+    rows: int
+    columns: int
+    num_channels: int
+    rows_axis: int
+    columns_axis: int
+    num_channels_axis: int
+    frame_axis: int
+
+    def __post_init__(self) -> None:
+        self._validate_video_structure()
+        self._initialize_frame_shape()
+        self.number_of_pixels_per_frame = np.prod(self.frame_shape)
+
+    def _initialize_frame_shape(self) -> None:
+        self.frame_shape = [None, None, None, None]
+        self.frame_shape[self.rows_axis] = self.rows
+        self.frame_shape[self.columns_axis] = self.columns
+        self.frame_shape[self.num_channels_axis] = self.num_channels
+        self.frame_shape.pop(self.frame_axis)
+        self.frame_shape = tuple(self.frame_shape)
+
+    def _validate_video_structure(self) -> None:
+
+        exception_message = (
+            "Invalid structure: "
+            f"{self.__repr__()}, "
+            "each property axis should be unique value between 0 and 3 (inclusive)"
+        )
+
+        axis_values = set((self.rows_axis, self.columns_axis, self.num_channels_axis, self.frame_axis))
+        axis_values_are_not_unique = len(axis_values) != 4
+        if axis_values_are_not_unique:
+            raise ValueError(exception_message)
+
+        values_out_of_range = any([axis < 0 or axis > 4 for axis in axis_values])
+        if values_out_of_range:
+            raise ValueError(exception_message)
+
+    def build_video_shape(self, n_frames: int) -> Tuple[int, int, int, int]:
+        video_shape = [None, None, None, None]
```

Comment:
```suggestion
        video_shape = [None] * 4
```

Simpler way of defining a list of homogenous items of specified length